presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeIdenticalWindows.java

```diff
@@ -0,0 +1,150 @@
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "quantity"), functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                        any()))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsABcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by orderkey order by shipdate rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("lag", "quantity", "*", "*")),
+                                        project(
+                                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                                        any()))))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsAAcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by suppkey order by orderkey rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount"), functionCall("lag", "quantity", "*", "*")),
+                                project(
+                                        window(ImmutableList.of(functionCall("sum", "quantity")),
+                                                any())))));
+    }
+
+    private void assertUnitPlan(@Language("SQL") String sql, PlanMatchPattern pattern)
+    {
+        Plan actualPlan = unitPlan(sql);
+        queryRunner.inTransaction(transactionSession -> {
+            PlanAssert.assertPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    private Plan unitPlan(@Language("SQL") String sql)
```

Comment:
I see your point. Nevertheless, I think you still need to have a test which uses the same planner configuration as it used in production to avoid "false positives" (when the test passes, but functionality does not work on production). It would be sad having TMIW passing and this rewrite was not being used in production. This kind of bug is hard to find as this is an optimization rewrite, query passes anyway (with or without the rewrite). Product tests won't catch it for sure.

From the other side, the chance that other rule would break the result of MIW, or that the MIW is not used seems to be low. I am not going argue about this comment. I shared my concern, but it is you who makes the decision.